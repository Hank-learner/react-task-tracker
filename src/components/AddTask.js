import { useState } from 'react'

const AddTask = ({ onAddTask, showAddTask }) => {

    const [text, setText] = useState('')
    const [day, setDay] = useState('')
    const [reminder, setReminder] = useState(false)

    const onSubmit = (e) => {
        e.preventDefault();
        if (!text) {
            alert('Please add task')
            return
        }
        onAddTask({ text, day, reminder })

        setText('')
        setDay('')
        setReminder(false)
    }

    return (
            showAddTask ? (
                <form className='add-form' onSubmit={onSubmit} >
                    <div className='form-control'>
                        <label htmlFor='new-task-title'>Enter Task</label>
                        <input
                            id='new-task-title'
                            type='text'
                            placeholder='Add Task'
                            value={text}
                            onChange={(e) => setText(e.target.value)}
                            required
                        />
                    </div>
                    <div className='form-control'>
                        <label>Day & Time</label>
                        <input
                            type='text'
                            placeholder='Add Day & Time'
                            value={day}
                            onChange={(e) => setDay(e.target.value)}
                        />
                    </div>
                    <div className='form-control form-control-check'>
                        <label>Set Reminder</label>
                        <input
                            type='checkbox'
                            checked={reminder}
                            value={reminder}
                            onChange={(e) => setReminder(e.currentTarget.checked)}
                        />
                    </div>
                    <input
                        type='submit'
                        className='btn btn-block'
                        value='Save Task'
                />
                </form >
            ):(null) 
    )
};

export default AddTask
