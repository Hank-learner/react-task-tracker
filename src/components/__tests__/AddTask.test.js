import { render, screen, getQueriesForElement } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AddTask from '../AddTask';
import App from '../../App';

// const validProps = {

// }


describe('AddTask snapshot test', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<AddTask {...App.addTask} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('render testing', () => {

    test('renders addTask', () => {
        const { getByLabelText } = render(<AddTask {...App.addTask} />);
        expect(getByLabelText('Enter Task')).not.toBeNull();
    });
})

//  const {getByTestId} = render(<ContactForm {...props} />);
//  const contactInput = getByTestId(dataLocators.contact_name);
//  expect(contactInput).toBeDefined();
//  fireEvent(contactInput, 'onBlur');
//  expect(contactInput.props.value).toBe('gaurav++==');
//  fireEvent(
//  contactInput,
//  'onChangeText',
//  'really long long long long name dfbdjfbdfjgbdf fgkbjdfgfg',
//  );
//  expect(props.onChangeHandler).toBeCalledTimes(1);
//  });