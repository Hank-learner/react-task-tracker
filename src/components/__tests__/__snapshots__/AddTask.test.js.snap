// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask snapshot test renders correctly 1`] = `
<form
  className="add-form"
  onSubmit={[Function]}
>
  <div
    className="form-control"
  >
    <label
      htmlFor="new-task-title"
    >
      Enter Task
    </label>
    <input
      id="new-task-title"
      onChange={[Function]}
      placeholder="Add Task"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-control"
  >
    <label>
      Day & Time
    </label>
    <input
      onChange={[Function]}
      placeholder="Add Day & Time"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-control form-control-check"
  >
    <label>
      Set Reminder
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </div>
  <input
    className="btn btn-block"
    type="submit"
    value="Save Task"
  />
</form>
`;
